libbeat/docs/outputs/output-logstash.asciidoc

```diff
@@ -0,0 +1,397 @@
+[[logstash-output]]
+=== Configure the Logstash output
+
+++++
+<titleabbrev>Logstash</titleabbrev>
+++++
+
+The Logstash output sends events directly to Logstash by using the lumberjack
+protocol, which runs over TCP. Logstash allows for additional processing and routing of
+generated events.
+
+// tag::shared-logstash-config[]
+
+[IMPORTANT]
+.Prerequisite
+To send events to {ls}, you also need to create a {ls} configuration pipeline
+that listens for incoming Beats connections and indexes the received events into
+{es}. For more information, see the section about
+{stack-gs}/get-started-elastic-stack.html#logstash-setup[configuring {ls}] in
+the {stack} getting started tutorial. Also see the documentation for the
+{logstash-ref}/plugins-inputs-beats.html[{beats} input] and
+{logstash-ref}/plugins-outputs-elasticsearch.html[{es} output] plugins.
+
+If you want to use {ls} to perform additional processing on the data collected by
+{beatname_uc}, you need to configure {beatname_uc} to use {ls}.
+
+To do this, you edit the {beatname_uc} configuration file to disable the {es}
+output by commenting it out and enable the {ls} output by uncommenting the
+logstash section:
+
+[source,yaml]
+------------------------------------------------------------------------------
+#----------------------------- Logstash output --------------------------------
+output.logstash:
+  hosts: ["127.0.0.1:5044"]
+------------------------------------------------------------------------------
+
+The `hosts` option specifies the {ls} server and the port (`5044`) where {ls} is configured to listen for incoming
+Beats connections.
+
+For this configuration, you must <<load-template-manually,load the index template into {es} manually>>
+because the options for auto loading the template are only available for the {es} output.
+
+ifeval::["{beatname_lc}"=="filebeat"]
+Want to use <<filebeat-modules,{beatname_uc} modules>> with {ls}? You need to do
+some extra setup. For more information, see
+{logstash-ref}/filebeat-modules.html[Working with {beatname_uc} modules].
+endif::[]
+
+ifndef::win-only[]
+ifndef::apm-server[]
+ 
+include::{libbeat-dir}/docs/step-test-config.asciidoc[]
+
+endif::apm-server[]
+ifdef::apm-server[]
+
+include::./step-test-config.asciidoc[]
```

Comment:
Because this file is now one level deeper, can you please update this include for APM Server:

```suggestion
include::./../step-test-config.asciidoc[]
```

Reply:
actually, now that we're using `{docdir}` to resolve the path to  libbeat, I can remove the conditional coding, and it should work for APM, too. You'll just need to modify your setting for {libbeat-dir}. If this causes problems, we can open another PR to use the conditional statement.